crane-check/main.go

```diff
@@ -0,0 +1,234 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Throw an error early if we provide an invalid match mode
+			if !isValidMatchMode(cfg.MatchMode) {
+				return fmt.Errorf("invalid match-mode: %s (must be one of: %s)", cfg.MatchMode, strings.Join(validMatchModes, ", "))
+			}
+			return cfg.Run(cmd.Context())
+		},
+	}
+
+	cmd.Flags().StringVar(&cfg.Image, "image", "", "Image to be assessed (e.g., cgr.dev/chainguard/crane:latest) [required]")
+	cmd.Flags().StringVar(&cfg.Environment, "env", "", "Environment variables to check (space or newline-separated, format: ENV_VAR=VALUE) [required]")
+	cmd.Flags().StringVar(&cfg.MatchMode, "match", "exact", "Match mode: 'exact', 'relative', or 'contains'")
+
+	cmd.MarkFlagRequired("image")
+	cmd.MarkFlagRequired("env")
+
+	return cmd
+}
+
+func (c *cfg) Run(ctx context.Context) error {
+	log := clog.FromContext(ctx).With("image", c.Image)
+
+	// Parse passed environment variables
+	envVars := parseEnvironmentVariables(c.Environment)
+	if envVars == nil {
+		return fmt.Errorf("no environment variables provided")
+	}
+	log.InfoContext(ctx, "retrieving image configuration")
+
+	// Get the image config with crane
+	imageConfig, err := getCraneConfig(ctx, c.Image)
+	if err != nil {
+		return fmt.Errorf("failed to get image configuration: %w", err)
+	}
+
+	// Parse the config
+	var imgCfg ImageConfig
+	if err := json.Unmarshal([]byte(imageConfig), &imgCfg); err != nil {
+		return fmt.Errorf("failed to parse image configuration: %w", err)
+	}
+
+	// Ensure environment exists in config
+	if imgCfg.Config.Env == nil {
+		return fmt.Errorf("failed to parse environment from image config")
+	}
+	log.InfoContext(ctx, "successfully parsed environment from image config")
+
+	// Check each environment variable
+	for _, envVar := range envVars {
+		if err := matchEnvironmentVariable(ctx, envVar, imgCfg.Config.Env, c.MatchMode); err != nil {
+			return err
+		}
+	}
+	log.InfoContext(ctx, "all environment variables match expected values")
+
+	return nil
+}
+
+func parseEnvironmentVariables(envStr string) []string {
+	envStr = strings.TrimSpace(envStr)
+	if envStr == "" {
+		return nil
+	}
+
+	// Attempt to split input via newlines first
+	// I want newline separation to "just" work in our pipelines
+	lines := strings.Split(envStr, "\n")
+	var envVars []string
+
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+
+		// Ignore commented out lines and skip empty ones
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// Account for space separated env vars
+		if strings.Contains(line, " ") && strings.Count(line, "=") > 1 {
+			envs := strings.Fields(line)
+			envVars = append(envVars, envs...)
+		} else {
+			envVars = append(envVars, line)
+		}
+	}
+
+	return envVars
+}
+
+func getCraneConfig(ctx context.Context, image string) (string, error) {
+	log := clog.FromContext(ctx)
+
+	// Get image config with crane
+	configJSON, err := crane.Config(image, crane.WithContext(ctx))
```

Comment:
This will always be checking the `amd64` image by default, which we might be okay with? We could plumb an arch flag down to this as well in case we want to check the `arm64` variant?